unstructured/ingest/connector/astra.py

```diff
@@ -111,7 +133,89 @@ def normalize_dict(self, element_dict: dict) -> dict:
         return {
             "$vector": element_dict.pop("embeddings", None),
             "content": element_dict.pop("text", None),
-            "metadata": flatten_dict(
-                element_dict, separator="-", flatten_lists=True, remove_none=True
-            ),
+            "metadata": element_dict,
         }
+
+@dataclass
+class AstraIngestDoc(IngestDocCleanupMixin, BaseSingleIngestDoc):
+    connector_config: SimpleAstraConfig
+    meta: t.Dict[str, str] = field(default_factory=dict)
+    registry_name: str = "astra"
+
+    def update_source_metadata(self):
+        self.source_metadata = SourceMetadata(
+            exists=True,
+        )
```

Comment:
@erichare Could you look deeper into this? Best example might be delta_table.py since it is also reading from a database.

Reply:
Got it. Yes, this is a great example - basically pulling the data into Pandas dataframes, writing them as CSVs, which presumably then become the input which gets converted into JSON for Unstructured.

I think we can follow the exact paradigm with minor changes, and i've done that (will push shortly)... I don't know the general protocol here but I would be happy also to credit them in some way with the inspiration for that solution